### PR TITLE
feat: [CDS-35302]: ssh connector creation wizard

### PR DESCRIPTION
**Pull request by azharness**
##### Screenshots:
Screenshot 2022-05-23 at 16 45 05
Screenshot 2022-05-23 at 16 45 12
Screenshot 2022-05-23 at 16 45 22
Screenshot 2022-05-23 at 16 45 33
Screenshot 2022-05-23 at 16 45 46

You can use the following comments to re-trigger PR Checks

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier - `fix prettier`

### CODE REVIEW

File: src/modules/35-connectors/components/CreateConnector/PdcConnector/components/UploadJSON.tsx [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+      onDrop={event => {
+        preventDefaults(event)
+        setDropHighlight(false)
+        if (event.dataTransfer.items) {
+          // Use DataTransferItemList interface to access the file(s)
+          for (let i = 0; i < event.dataTransfer.items.length; i++) {
+            // If dropped items aren't files, reject them
+            if (event.dataTransfer.items[i].kind === 'file') {
+              const file = event.dataTransfer.items[i].getAsFile()
+              handleFileUpload(file as any)
```
Comment (diptiman-adak) [inline src/modules/35-connectors/components/CreateConnector/PdcConnector/components/UploadJSON.tsx:80]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/harness/harness-core-ui/pullRequest?prid=9232306)
```suggestion
              handleFileUpload(file as unknown)
```

File: src/modules/35-connectors/components/CreateConnector/PdcConnector/StepDetails/PdcDetails.tsx [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+
+  useEffect(() => {
+    setManualTypedHosts(prevStepData?.hosts || '')
+  }, [])
+
+  useEffect(() => {
+    if (loadingConnectorSecrets) {
+      if (props.isEditMode) {
+        if (props.connectorInfo) {
+          setManualTypedHosts(props.connectorInfo?.spec?.hosts?.map?.((host: any) => host.hostname).join('\n'))
```
Comment (diptiman-adak) [inline src/modules/35-connectors/components/CreateConnector/PdcConnector/StepDetails/PdcDetails.tsx:73]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/harness/harness-core-ui/pullRequest?prid=9232306)
```suggestion
          setManualTypedHosts(props.connectorInfo?.spec?.hosts?.map?.((host: unknown) => host.hostname).join('\n'))
```

File: src/modules/35-connectors/components/CreateConnector/PdcConnector/components/UploadJSON.tsx [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+        }
+      }}
+    >
+      <input
+        type="file"
+        id="bulk"
+        name="bulk-upload"
+        style={{ display: 'none' }}
+        ref={inputRef}
+        onChange={event => handleFileUpload((event.target as any).files[0])}
```
Comment (diptiman-adak) [inline src/modules/35-connectors/components/CreateConnector/PdcConnector/components/UploadJSON.tsx:97]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/harness/harness-core-ui/pullRequest?prid=9232306)
```suggestion
        onChange={event => handleFileUpload((event.target as unknown).files[0])}
```

File: src/modules/35-connectors/components/CreateConnector/PdcConnector/components/UploadJSON.tsx [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+  const { getString } = useStrings()
+  const { showError } = useToaster()
+  const inputRef = useRef() as React.MutableRefObject<HTMLInputElement>
+  const [fileContent, setFileContent] = useState([] as uploadHostItem[])
+  const [fileName, setFileName] = useState<string>('')
+  const [dropHighlight, setDropHighlight] = useState(false)
+
+  useEffect(() => {
+    setJsonValue(fileContent)
+  }, [fileContent])
```
Comment (diptiman-adak) [inline src/modules/35-connectors/components/CreateConnector/PdcConnector/components/UploadJSON.tsx:30]: Codacy has a fix for the issue: [React Hook useEffect has a missing dependency: 'setJsonValue'. Either include it or remove the dependency array. If 'setJsonValue' changes too often, find the parent component that defines it and wrap that definition in useCallback.](https://app.codacy.com/gh/harness/harness-core-ui/pullRequest?prid=9232306)
```suggestion
  }, [fileContent, setJsonValue])
```

File: src/modules/35-connectors/components/CreateConnector/PdcConnector/StepDetails/PdcDetails.tsx [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+  const defaultInitialFormData: PDCFormInterface = {}
+
+  const [initialValues, setInitialValues] = useState(defaultInitialFormData)
+  const [hostsJSON, setHostsJSON] = useState([] as uploadHostItem[])
+  const [loadingConnectorSecrets, setLoadingConnectorSecrets] = useState(props.isEditMode)
+  const [manualTypedHosts, setManualTypedHosts] = useState('')
+
+  useEffect(() => {
+    setManualTypedHosts(prevStepData?.hosts || '')
+  }, [])
```
Comment (diptiman-adak) [inline src/modules/35-connectors/components/CreateConnector/PdcConnector/StepDetails/PdcDetails.tsx:67]: Codacy has a fix for the issue: [React Hook useEffect has a missing dependency: 'prevStepData?.hosts'. Either include it or remove the dependency array.](https://app.codacy.com/gh/harness/harness-core-ui/pullRequest?prid=9232306)
```suggestion
  }, [prevStepData?.hosts])
```

File: src/modules/35-connectors/components/CreateConnector/PdcConnector/components/UploadJSON.tsx (hunk 1 above)
Comment (diptiman-adak) [inline src/modules/35-connectors/components/CreateConnector/PdcConnector/components/UploadJSON.tsx:80]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/harness/harness-core-ui/pullRequest?prid=9232306)
```suggestion
              handleFileUpload(file as unknown)
```

File: src/modules/35-connectors/components/CreateConnector/PdcConnector/StepDetails/TestConnection.tsx [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+  const { getString } = useStrings()
+
+  const { prevStepData } = props
+  const { accountId, orgIdentifier, projectIdentifier } = useParams<ProjectPathProps>()
+  const [currentStatus, setCurrentStatus] = useState<Status>(Status.PROCESS)
+  const [currentIntent, setCurrentIntent] = useState<Intent>(Intent.NONE)
+  const [currentStep, setCurrentStep] = useState<number>(1)
+  const [errors, setErrors] = useState<ResponseMessage[] | null>(null)
+
+  const steps: string[] = useMemo(() => [getString('connectors.pdc.testConnection.step1')], [])
```
Comment (diptiman-adak) [inline src/modules/35-connectors/components/CreateConnector/PdcConnector/StepDetails/TestConnection.tsx:51]: Codacy has a fix for the issue: [React Hook useMemo has a missing dependency: 'getString'. Either include it or remove the dependency array.](https://app.codacy.com/gh/harness/harness-core-ui/pullRequest?prid=9232306)
```suggestion
  const steps: string[] = useMemo(() => [getString('connectors.pdc.testConnection.step1')], [getString])
```

File: src/modules/35-connectors/components/CreateConnector/PdcConnector/StepDetails/PdcDetails.tsx (hunk 5 above)
Comment (diptiman-adak) [inline src/modules/35-connectors/components/CreateConnector/PdcConnector/StepDetails/PdcDetails.tsx:67]: Codacy has a fix for the issue: [React Hook useEffect has a missing dependency: 'prevStepData?.hosts'. Either include it or remove the dependency array.](https://app.codacy.com/gh/harness/harness-core-ui/pullRequest?prid=9232306)
```suggestion
  }, [prevStepData?.hosts])
```

File: src/modules/35-connectors/components/CreateConnector/PdcConnector/StepDetails/TestConnection.tsx [hunk 7]
```diff
@@ -0,0 +1,127 @@
...
+      setCurrentStatus(Status.ERROR)
+      setCurrentIntent(Intent.DANGER)
+    }
+  }
+
+  useEffect(() => {
+    setTimeout(() => {
+      verifyTestConnection()
+    }, 200)
+  }, [])
```
Comment (diptiman-adak) [inline src/modules/35-connectors/components/CreateConnector/PdcConnector/StepDetails/TestConnection.tsx:96]: Codacy has a fix for the issue: [React Hook useEffect has a missing dependency: 'verifyTestConnection'. Either include it or remove the dependency array.](https://app.codacy.com/gh/harness/harness-core-ui/pullRequest?prid=9232306)
```suggestion
  }, [verifyTestConnection])
```

File: src/modules/35-connectors/components/CreateConnector/PdcConnector/StepDetails/TestConnection.tsx [hunk 8]
```diff
@@ -0,0 +1,130 @@
...
+            message: `${error.reason}. ${error.message}`
+          })) as ResponseMessage[]
+        )
+      }
+    } catch (e) {
+      setErrors(e.data?.responseMessages)
+      setCurrentStatus(Status.ERROR)
+      setCurrentIntent(Intent.DANGER)
+    }
+  }, [])
```
Comment (diptiman-adak) [inline src/modules/35-connectors/components/CreateConnector/PdcConnector/StepDetails/TestConnection.tsx:91]: Codacy has a fix for the issue: [React Hook useEffect has a missing dependency: 'verifyTestConnection'. Either include it or remove the dependency array.](https://app.codacy.com/gh/harness/harness-core-ui/pullRequest?prid=9232306)
```suggestion
  }, [verifyTestConnection])
```

File: src/modules/35-connectors/components/CreateConnector/PdcConnector/components/UploadJSON.tsx [hunk 9]
```diff
@@ -0,0 +1,107 @@
...
+      onDrop={event => {
+        preventDefaults(event)
+        setDropHighlight(false)
+        if (event.dataTransfer.items) {
+          // Use DataTransferItemList interface to access the file(s)
+          for (let i = 0; i < event.dataTransfer.items.length; i++) {
+            // If dropped items aren't files, reject them
+            if (event.dataTransfer.items[i].kind === 'file') {
+              const file = event.dataTransfer.items[i].getAsFile()
+              handleFileUpload(file as any)
```
Comment (diptiman-adak) [inline src/modules/35-connectors/components/CreateConnector/PdcConnector/components/UploadJSON.tsx:75]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/harness/harness-core-ui/pullRequest?prid=9232306)
```suggestion
              handleFileUpload(file as unknown)
```

File: src/modules/35-connectors/components/CreateConnector/PdcConnector/StepDetails/PdcDetails.tsx [hunk 10]
```diff
@@ -0,0 +1,131 @@
...
+  const { prevStepData, nextStep } = props
+  const { getString } = useStrings()
+
+  const [hostsJSON, setHostsJSON] = useState([] as uploadHostItem[])
+
+  useEffect(() => {
+    if (typeof prevStepData?.spec?.hosts === 'object') {
+      setHostsJSON(prevStepData?.spec?.hosts)
+    }
+  }, [])
```
Comment (diptiman-adak) [inline src/modules/35-connectors/components/CreateConnector/PdcConnector/StepDetails/PdcDetails.tsx:60]: Codacy has a fix for the issue: [React Hook useEffect has a missing dependency: 'prevStepData?.hosts'. Either include it or remove the dependency array.](https://app.codacy.com/gh/harness/harness-core-ui/pullRequest?prid=9232306)
```suggestion
  }, [prevStepData?.hosts])
```

File: src/modules/35-connectors/components/CreateConnector/PdcConnector/StepDetails/PdcDetails.tsx [hunk 11]
```diff
@@ -0,0 +1,131 @@
...
+    if (hostsJSON?.length > 0) {
+      data.hosts = hostsJSON
+    }
+    nextStep?.({ ...prevStepData, ...data } as StepConfigureProps)
+  }
+
+  const initialFormValues = useMemo(() => {
+    const hosts = prevStepData?.hosts || prevStepData?.spec?.hosts
+    return {
+      hosts: typeof hosts === 'string' ? hosts : hosts?.map?.((host: any) => host.hostname).join('\n')
```
Comment (diptiman-adak) [inline src/modules/35-connectors/components/CreateConnector/PdcConnector/StepDetails/PdcDetails.tsx:73]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/harness/harness-core-ui/pullRequest?prid=9232306)
```suggestion
      hosts: typeof hosts === 'string' ? hosts : hosts?.map?.((host: unknown) => host.hostname).join('\n')
```

File: src/modules/35-connectors/components/CreateConnector/PdcConnector/components/UploadJSON.tsx [hunk 12]
```diff
@@ -0,0 +1,107 @@
...
+        }
+      }}
+    >
+      <input
+        type="file"
+        id="bulk"
+        name="bulk-upload"
+        style={{ display: 'none' }}
+        ref={inputRef}
+        onChange={event => handleFileUpload((event.target as any).files[0])}
```
Comment (diptiman-adak) [inline src/modules/35-connectors/components/CreateConnector/PdcConnector/components/UploadJSON.tsx:92]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/harness/harness-core-ui/pullRequest?prid=9232306)
```suggestion
        onChange={event => handleFileUpload((event.target as unknown).files[0])}
```

Comment (diptiman-adak) [inline src/modules/35-connectors/components/CreateConnector/PdcConnector/components/UploadJSON.tsx:92]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/harness/harness-core-ui/pullRequest?prid=9232306)
```suggestion
        onChange={event => handleFileUpload((event.target as unknown).files[0])}
```

File: src/modules/35-connectors/components/CreateConnector/PdcConnector/StepDetails/TestConnection.tsx [hunk 13]
```diff
@@ -0,0 +1,117 @@
...
```
Comment (diptiman-adak) [inline src/modules/35-connectors/components/CreateConnector/PdcConnector/StepDetails/TestConnection.tsx:91]: Codacy has a fix for the issue: [React Hook useEffect has a missing dependency: 'verifyTestConnection'. Either include it or remove the dependency array.](https://app.codacy.com/gh/harness/harness-core-ui/pullRequest?prid=9232306)
```suggestion
  }, [verifyTestConnection])
```

File: src/modules/35-connectors/components/CreateConnector/PdcConnector/StepDetails/PdcDetails.tsx [hunk 14]
```diff
@@ -0,0 +1,130 @@
...
```
Comment (diptiman-adak) [inline src/modules/35-connectors/components/CreateConnector/PdcConnector/StepDetails/PdcDetails.tsx:72]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/harness/harness-core-ui/pullRequest?prid=9232306)
```suggestion
      hosts: typeof hosts === 'string' ? hosts : hosts?.map?.((host: unknown) => host.hostname).join('\n')
```

File: src/modules/35-connectors/components/CreateConnector/PdcConnector/components/UploadJSON.tsx (hunk 12 above)
Comment (diptiman-adak) [inline src/modules/35-connectors/components/CreateConnector/PdcConnector/components/UploadJSON.tsx:92]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/harness/harness-core-ui/pullRequest?prid=9232306)
```suggestion
        onChange={event => handleFileUpload((event.target as unknown).files[0])}
```

File: src/modules/35-connectors/components/CreateConnector/PdcConnector/StepDetails/TestConnection.tsx [hunk 15]
```diff
@@ -0,0 +1,129 @@
...
```
Comment (diptiman-adak) [inline src/modules/35-connectors/components/CreateConnector/PdcConnector/StepDetails/TestConnection.tsx:91]: Codacy has a fix for the issue: [React Hook useEffect has a missing dependency: 'verifyTestConnection'. Either include it or remove the dependency array.](https://app.codacy.com/gh/harness/harness-core-ui/pullRequest?prid=9232306)
```suggestion
  }, [verifyTestConnection])
```

**Comment by azharness** [comment]
retrigger jest

**Comment by azharness** [comment]
retrigger jest

**Comment by azharness** [comment]
retrigger coverage

**Comment by azharness** [comment]
retrigger coverage

**Comment by azharness** [comment]
retrigger jest

**Comment by azharness** [comment]
retrigger coverage

**Comment by azharness** [comment]
retrigger jest

**Comment by azharness** [comment]
retrigger coverage

**Comment by azharness** [comment]
retrigger coverage

**Comment by azharness** [comment]
retrigger jest

**Comment by azharness** [comment]
retrigger coverage

**Comment by azharness** [comment]
retrigger coverage

**Comment by azharness** [comment]
retrigger jest

**Comment by azharness** [comment]
retrigger coverage

**Comment by azharness** [comment]
retrigger jest

**Review by Tamaraius — APPROVED** [review]

**Comment by azharness** [comment]
retrigger jest

**Comment by azharness** [comment]
trigger cypress

**Comment by azharness** [comment]
retrigger cypress

**Comment by azharness** [comment]
retrigger jest

**Review by Tamaraius — APPROVED** [review]

**Comment by azharness** [comment]
trigger rebase

**Comment by azharness** [comment]
trigger rebase

**Comment by azharness** [comment]
retrigger jest

**Comment by azharness** [comment]
retrigger cypress

**Comment by azharness** [comment]
retrigger jest

**Review by Tamaraius — DISMISSED** [review]

**Review by kapilkumawat86 — CHANGES_REQUESTED** [review]
Fix cypress test case, since its broken after your change only

**Comment by azharness** [comment]
trigger cypress

**Comment by azharness** [comment]
retrigger jest

**Comment by azharness** [comment]
retrigger jest

**Pull request merged**